addons/mass_mailing/static/src/snippets/s_social_media_mass_mailing/options.js

```diff
@@ -0,0 +1,109 @@
+const rpc = require('web.rpc');
+const session = require('web.session');
+
+let dbSocialValues;
+let dbSocialValuesProm;
+options.registry.SocialMediaMassMailing = SocialMedia.extend({
+    /**
+     * @override
+     */
+    async cleanForSave() {
+        await this._fetchSocialMedia();
+        const dbSocialValuesToSave = new Set();
+        for (const [mediaName, url] of Object.entries(dbSocialValues)) {
+            if (!url) {
+                dbSocialValuesToSave.add(mediaName);
+            }
+        }
+        const dict = {};
+        for (const entry of this._lastEntries) {
+            if (entry.media && entry.display_name && dbSocialValuesToSave.has(entry.media)) {
+                dbSocialValues[entry.media] = entry.display_name;
+                dict[`social_${entry.media}`] = entry.display_name;
+                dbSocialValuesToSave.delete(entry.media);
+            }
+        }
+
+        const query = rpc.buildQuery({
+            model: 'res.company',
+            method: 'write',
+            args: [[session.company_id], dict],
+        });
+        await session.rpc(query.route, query.params).catch(e => {
+            if (e.message.exceptionName !== 'odoo.exceptions.AccessError') {
+                return Promise.reject(e);
+            }
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    editCompanySocial(previewMode, widgetValue, params) {
+        if (previewMode !== false) {
+            return;
+        }
+        this.do_action({
+            res_id: session.company_id,
+            res_model: "res.company",
+            type: "ir.actions.act_window",
+            views: [[false, "form"]],
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    async _fetchSocialMedia() {
+        if (dbSocialValues) {
+            return dbSocialValues;
+        }
+        if (!dbSocialValuesProm) {
+            dbSocialValuesProm = this._rpc({
+                model: 'res.company',
+                method: 'read',
+                args: [[session.company_id], ['social_facebook', 'social_twitter', 'social_linkedin',
+                    'social_youtube', 'social_instagram', 'social_github']],
+            }).then(([values]) => {
+                delete values.id;
+                dbSocialValues = {};
+                for (const [name, url] of Object.entries(values)) {
+                    dbSocialValues[name.split('social_')[1]] = url;
+                }
+            });
+        }
+        await dbSocialValuesProm;
+    },
+    /**
+     * @override
+     */
+    async _getDbMedias() {
+        await this._fetchSocialMedia();
+        return {...dbSocialValues};
+    },
```

Comment:
Why do you need a duplicated method for `_fetchSocialMedia`? :open_mouth: Very curious, if it's needed, it needs a very informative comment :+1: